src/parser/utils.rs

```diff
@@ -361,3 +362,53 @@ pub fn strip_indent_doc(doc: String) -> String {
         .next()
         .expect("expected non-empty chunks after indentation of documentation")
 }
+
+pub fn check_unbound(types: Types) -> Result<Types, LexicalError> {
```

Comment:
It's a bit strange that `check_unbound` return `Types` in the success case. It should probably borrows immutably the `types` and return a `Result<(), LexicalError>` instead.